File: java/src/test/java/problems401_500/IslandPerimeterTest.java
package problems401_500;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class IslandPerimeterTest extends SampledTest {
    @Test
    public void test() {
        for (IslandPerimeter solution : IslandPerimeter.values()) {
            for (IslandPerimeterSample sample : getSamples(IslandPerimeterSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.islandPerimeter(sample.input.grid))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems401_500/IslandPerimeterTest.kt
package problems401_500

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class IslandPerimeterTest : SampledTest() {
    @Test
    fun test() =
        IslandPerimeter.entries.forEach { solution ->
            getSamples<Array<IslandPerimeterSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.islandPerimeter(input.grid))
                    .isEqualTo(output)
            }
        }
}
